src/LottoGame.js

```diff
@@ -0,0 +1,91 @@
+import { Console } from "@woowacourse/mission-utils";
+import LottoManager from "./LottoManager.js";
+import { INPUT_MESSAGE, PRINT_MESSAGE } from "./constants/lottoMessage.js";
+import { LOTTO_PRIZE } from "./constants/lottoRule.js";
+
+class LottoGame {
+  constructor() {
+    this.lottoManager = new LottoManager();
+  }
+
+  async start() {
+    const purchaseAmount = await this.getPurchaseAmount();
+    const lottos = this.lottoManager.buyLottos(purchaseAmount);
+    this.printPurchasedLottos(lottos);
+
+    const winningNumbers = await this.getWinningNumbers();
+    const bonusNumber = await this.getBonusNumber(winningNumbers);
+
+    const result = this.lottoManager.checkWinningResult(
+      lottos,
+      winningNumbers,
+      bonusNumber
+    );
+    this.printWinningStatistics(result, purchaseAmount);
+  }
+
+  async getPurchaseAmount() {
+    while (true) {
+      try {
+        const input = await Console.readLineAsync(
+          INPUT_MESSAGE.PURCHASE_AMOUNT
+        );
+        const amount = parseInt(input);
+        this.lottoManager.validatePurchaseAmount(amount);
+        return amount;
+      } catch (error) {
+        Console.print(error.message);
+      }
+    }
+  }
```

Comment:
여기서 `while`문을 사용하는 것 보다 catch에서 `this.getPurchaseAmount()`를 다시 실행시키는 방법은 어떨까요?
```suggestion
  async getPurchaseAmount() {
    try {
      const input = await Console.readLineAsync(
        INPUT_MESSAGE.PURCHASE_AMOUNT
      );
      const amount = parseInt(input);
      this.lottoManager.validatePurchaseAmount(amount);
      return amount;
    } catch (error) {
      Console.print(error.message);
      await this.getPurchaseAmount();
    }
  }
```

뒤에서 사용하신 `while`문들 모두 같은 방법으로 제거할 수 있을 것 같아요! 이렇게 하면 depth도 줄일 수 있고 좋답니다 👍

Reply:
오  ! 이런 방법이 이게 훨씬 좋은 것 같아요 !!

Reply:
재귀적으로 계속 호출하는 것은 스택 오버 플로우가 발생해 성능에 문제가 갈 수 있다고 하더라구요!

그래서 저도 catch문에서 다시 호출할까 하다가 while문을 사용해서 루프를 돌았던 것 같아요🤔

Reply:
오.. 그런 일이...  새로운걸 또 알아가네요 !! 감사합니당